### ISSUE TITLE
Kolibri might be busy - wait a few moments and then reload this page

### ISSUE THREAD

**Issue by miyer03** [learningequality/kolibri#5753]
### Observed behavior

On installing Kolibri 0.12.5 and then starting it, its shows Kolibri Busy , Nginx 502 bad gateway. One pattern observed is that, after switching on the server if we wait for few minutues, this error doesn't seem to occur. If we do a Kolibri stop on terminal, it shows 

```
ERROR    During graceful shutdown, server says: Failed to start (check log file: /home/kolibri/.kolibri/logs/daemon.txt) (6)
ERROR    Not responding, killing with force 
```
It works if we do 
```
Kolibri Stop, 
Kolibri Start and 
sudo service nginx restart.
```

### Expected behavior
It should not show Kolibri Busy but should open the Kolibri page.

### User-facing consequences

When it shows Kolibri busy, the teachers will get confused as they won't know what needs to be done. This will intern impact classrooms.

### Errors and logs

Attached

### Steps to reproduce

Start the system and open Kolibri on port 8008.

### Context

 * Kolibri version : 0.12.5
 * Operating system : U.058buntu 14.04, 16.04 and 1
 * Browser : Chrome

![Screenshot from 2019-07-08 12-10-09](https://user-images.githubusercontent.com/8583367/60910814-6b95f400-a29f-11e9-84cd-41f5a519edcf.png)
![Screenshot from 2019-07-09 14-39-57](https://user-images.githubusercontent.com/8583367/60910815-6c2e8a80-a29f-11e9-8087-52e539f4ba13.png)
![Screenshot from 2019-07-09 23-01-35](https://user-images.githubusercontent.com/8583367/60910816-6c2e8a80-a29f-11e9-8b48-d804d73fc71e.png)
![Screenshot from 2019-07-09 23-01-41](https://user-images.githubusercontent.com/8583367/60910817-6c2e8a80-a29f-11e9-9147-42a290a4d66e.png)
![Screenshot from 2019-07-09 23-07-32](https://user-images.githubusercontent.com/8583367/60910818-6c2e8a80-a29f-11e9-9993-11ec288d89f5.png)

**Comment by rtibbles** [comment]
For context from our conversations off github - it seems like the Kolibri server does eventually start, but that it is taking a very long time for the server to be available on its port.

**Comment by indirectlylit** [comment]
Is this using the `kolibri-server` package or a custom Nginx config?

**Comment by rtibbles** [comment]
This is a custom nginx configuration. (but one that the kolibri-server nginx configuration is based on)

**Comment by indirectlylit** [comment]
there are two things we might want to look into then:

1. Is Kolibri taking longer than normal to start up, or is it just the standard amount of time?
2. Are there improvements we can make to the nginx "please wait" screen? I've created an issue for this here: https://github.com/learningequality/kolibri-server/issues/17  once that's done we can help port those changes back to this config

**Comment by intelliant01** [comment]
From my experience with the same while working with the kolibri-server
package, I can confirm that Kolibri is taking the "normal" time to startup
and the 502 error is shown while nginx waits upon it (basically vaccumming
of a large db.sqlite3 file).

My suggestion would be to improve the messaging by either displaying a more
clear message along with the last few lines of the kolibri.txt log.

The bothersome part here could be the error message during shutdown. Is
Kolibri actually shutting down cleanly? In most cases, I have noticed the
-shm and -wal files remaining behind even after a "shutdown".

**Comment by intelliant01** [comment]
> This is a custom nginx configuration.

The experience is the same even with the default kolibri-server installation with a large db albeit on port 8080.

**Comment by rtibbles** [comment]
Sounds like for a large database that this can take up to 30 minutes to complete, which is challenging for users.

**Comment by rtibbles** [comment]
@miyer03 if you run the following command:
`kolibri manage vacuumsqlite` before starting the server, does that cause any improvement?

**Comment by miyer03** [comment]
This seems to be showing up in 12.8 as well. If we wait for a couple of minutes before starting kolibri it goes away or once the page is refreshed after few minutes. It seemed not to occur using kolibri manage vacuumsqlite

**Comment by miyer03** [comment]
This seems to be showing up in 12.8 as well. If we wait or a couple of mins before starting kolibri it goes away or once the page is refreshed after a few mins. It seemed not to occur o using kolibri manage vacuumsqlite before starting kolibri. Config of the server

**Comment by miyer03** [comment]
This seems to be showing up in 12.8 as well. If we wait or a couple of mins before starting kolibri it goes away or once the page is refreshed after a few mins. It seemed not to occur o using kolibri manage vacuumsqlite before starting kolibri. Config of the server

Intel Celeron CPU N2840 @ 2.16 GHz x 2 Ubuntu 16.04 LTS

**Comment by indirectlylit** [comment]
This has been addressed by recent changes to the `kolibri-server` 502 error page:

* https://github.com/learningequality/kolibri-server/pull/35
* https://github.com/learningequality/kolibri-server/pull/36
* https://github.com/learningequality/kolibri-server/pull/40

and changes to startup behavior:

https://github.com/learningequality/kolibri/issues/5891

**Comment by indirectlylit** [comment]
Note that the startup behavior will be available in Kolibri 0.12.9, but the new `kolibri-server` loading page will be in 0.13.0
